### PR TITLE
ES6: Convert lib/detect

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `lib/detect` to ES6. Module needs a refactor ... In another PR I'll tackle to conversion of the Specs.

**TODO**

- [x] Fix Eslint
- [x] Fix commercial Specs
- [x] Go with the flow

### CODE REVIEW

File: static/src/javascripts/lib/detect.js [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+
+type MediaQueryList = {
+    matches: boolean,
+    media: string,
+    onchange: (callback: (event: Event) => void) => void,
+    addListener: (callback: (event: Event) => void) => void,
+    removeListener: (callback: (event: Event) => void) => void,
+};
+
+type BreakpointName =
```
Comment (sndrs) [inline static/src/javascripts/lib/detect.js]: ❤️💜💙💛💚

File: static/src/javascripts/lib/detect.js [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+
+        if (bp.mql) {
+            bp.mql.addListener(bp.listener);
+        }
+
+        bp.listener(bp.mql);
+    });
+};
+
+const init = (): void => {
```
Comment (sndrs) [inline static/src/javascripts/lib/detect.js]: maybe this should be `initBreakpoints` or similar?

**Comment by sndrs** [comment]
great stuff this! 

do you need to claim the file and remove the list too?

**Review by lps88 — COMMENTED** [review]
This is great! Keep up the good work 🏆

File: static/src/javascripts/projects/commercial/modules/third-party-tags/outbrain.spec.js [hunk 3]
```diff
@@ -32,8 +45,6 @@ describe('Outbrain', () => {
             section: 'uk-news',
             commentable: true,
         };
-
-        expect.hasAssertions();
```
Comment (lps88) [inline static/src/javascripts/projects/commercial/modules/third-party-tags/outbrain.spec.js]: Why is this removed? 😞

**Comment by gustavpursche** [comment]
@sndrs I think this is ready for a review, @sndrs @GHaberis. Would be great if you could have a look :)

File: tools/es5to6.json [hunk 4]
```diff
@@ -41,7 +41,6 @@
     "projects/common/modules/component.js"
   ],
   "Calum Campbell": [
-    "lib/detect.js",
```
Comment (sndrs) [inline tools/es5to6.json]: this also, for some weird reason, appears under my name too

**Review by sndrs — APPROVED** [review]
looks good to me (apart from comment). really good – this was a really nasty one!

File: static/src/javascripts/lib/detect.js [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+
+    if (breakpoint.isTweakpoint) {
+        currentBreakpoint = findBreakpoint(currentTweakpoint);
+    } else {
+        currentBreakpoint = currentTweakpoint;
+    }
+};
+
+// this function has a Breakpoint as context, so we can't use fat arrows
+const onMatchingBreakpoint = function(mql) {
```
Comment (GHaberis) [inline static/src/javascripts/lib/detect.js]: Could we add type annotation to onMatchingBreakpoint?

Comment (gustavpursche) [inline static/src/javascripts/lib/detect.js]: Oh, good catch.
